File: test_overpass.py
import requests
import json

overpass_url = "http://overpass-api.de/api/interpreter"
overpass_query = """
[out:json][timeout:60];
area["name"="Madhya Pradesh"]["admin_level"="4"]->.searchArea;
(
  node["amenity"="place_of_worship"]["religion"="jain"](area.searchArea);
  way["amenity"="place_of_worship"]["religion"="jain"](area.searchArea);
);
out center;
"""
try:
    response = requests.get(overpass_url, params={'data': overpass_query}, timeout=60)
    data = response.json()
    elements = data.get('elements', [])
    print(f"Found {len(elements)} Jain temples in MP.")
    if elements:
        print(elements[0])
except Exception as e:
    print(f"Error: {e}")

File: test_overpass.js
const query = `
[out:json][timeout:60];
area["name"="Madhya Pradesh"]["admin_level"="4"]->.searchArea;
(
  node["amenity"="place_of_worship"]["religion"="jain"](area.searchArea);
  way["amenity"="place_of_worship"]["religion"="jain"](area.searchArea);
);
out center;
`;

fetch(`http://overpass-api.de/api/interpreter`, {
  method: 'POST',
  body: 'data=' + encodeURIComponent(query),
  headers: {
    'Content-Type': 'application/x-www-form-urlencoded'
  }
})
.then(res => res.json())
.then(data => {
  const elements = data.elements || [];
  console.log(`Found ${elements.length} Jain temples in MP.`);
  if (elements.length > 0) {
    console.log(elements[0]);
  }
})
.catch(err => console.error("Error:", err));
